### PR TITLE
feat(Metis): add Metis Optimistic Layer2

### PR DESCRIPTION
**Pull request by developerfred**
### Metis Layer 2 

https://github.com/MetisProtocol

### Metis Dex 
- NetSwap
- Tethys
- Standard

### Contracts 
- [x] [Balance Checker](https://github.com/wbobeirne/eth-balance-checker)   [Andromeda Mainnet]()
- [x] [Balance Checker](https://github.com/wbobeirne/eth-balance-checker)   [Stardust](https://stardust-explorer.metis.io/address/0x2690906609A9AF3bF08B8384CD9A0F97A88A7E9c/contracts)

### Sourcify
- [ ] https://github.com/ethereum/sourcify/pull/621

### Contracts Testnet
[ITO](https://stardust-explorer.metis.io/address/0xdcA6F476EebCDE8FE8b072e3fC80dBC28dC209b3/contracts) ✅

[HappyRedPacket](https://stardust-explorer.metis.io/address/0xdB93cCd481012bB5D1E2c8d0aF7C5f2940c00fdC/contracts) ✅

[MaskBox](https://stardust-explorer.metis.io/address/0x97369fEE7db34E0BfE47861f2ec44b4378d13eB4/contracts) ✅

[HappyNFTRedPacket](https://stardust-explorer.metis.io/address/0x97369fEE7db34E0BfE47861f2ec44b4378d13eB4/transactions)✅

[BalanceCheck](https://stardust-explorer.metis.io/address/0x2690906609A9AF3bF08B8384CD9A0F97A88A7E9c/transactions)✅

### Metis Bridge 
https://bridge.metis.io/home

### Token List 
- https://github.com/DimensionDev/Mask-Token-List/pull/45

### Mask Assets 
- [ ]

### Mask contracts 

ITO - https://github.com/DimensionDev/InitialTwitterOffering/commit/1e351391384e3548193cb2032944eef477030b28
MysteryBox - https://github.com/DimensionDev/MysteryBox/tree/support_metis
RedPacket - https://github.com/DimensionDev/RedPacket/tree/support_metis

### CODE REVIEW

File: packages/mask/src/resources/NETSWAPIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { SvgIconProps, SvgIcon } from '@mui/material'
+
+const svg = (
+    <svg version="1.1" viewBox="0 0 26 26" width="26px" height="26px" xmlns="http://www.w3.org/2000/svg">
+        <g fill="none" fillRule="evenodd">
+            <g transform="translate(.99399 .000649)" fillRule="nonzero">
+                <g fill="#fff">
+                    <path
+                        id="形状结合"
```
Comment (Jack-Works) [inline packages/mask/src/resources/NETSWAPIcon.tsx]: Please remove those IDs.

File: packages/mask/src/plugins/EVM/assets/logo_white.016a0880.svg [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="111px" height="91px" viewBox="0 0 111 91" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <title>Untitled</title>
+    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <g id="logo" transform="translate(0.993993, 0.000649)" fill-rule="nonzero">
+            <g id="编组备份" fill="#FFFFFF">
```
Comment (Jack-Works) [inline packages/mask/src/plugins/EVM/assets/logo_white.016a0880.svg]: Please remove those IDs.

File: packages/mask/src/resources/NETSWAPIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { SvgIconProps, SvgIcon } from '@mui/material'
+
+const svg = (
+    <svg version="1.1" viewBox="0 0 26 26" width="26px" height="26px" xmlns="http://www.w3.org/2000/svg">
```
Comment (septs) [inline packages/mask/src/resources/NETSWAPIcon.tsx]: plz use https://aka.pw/svgomg reduce the svg image

Comment (developerfred) [reply packages/mask/src/resources/NETSWAPIcon.tsx]: ok!

**Review by Jack-Works — DISMISSED** [review]

File: packages/mask/src/components/shared/ApplicationBoard.tsx [hunk 4]
```diff
@@ -315,6 +316,7 @@ export function ApplicationBoard({ secondEntries, secondEntryChainTabs }: MaskAp
                         ChainId.Arbitrum,
                         ChainId.xDai,
                         ChainId.Celo,
+                        ChainId.Metis,
```
Comment (septs) [inline packages/mask/src/components/shared/ApplicationBoard.tsx]: https://github.com/DimensionDev/Maskbook/pull/5398

**Review by Jack-Works — DISMISSED** [review]

**Comment by developerfred** [comment]
**Dashboard**
- [x] Balance
Screenshot 2022-01-13 at 00 39 40
- [x] gas Price
Screenshot 2022-01-13 at 00 52 23

**Review by Jack-Works — DISMISSED** [review]

File: packages/web3-shared/evm/assets/chains.json [hunk 5]
```diff
@@ -196,28 +196,54 @@
         "infoURL": "https://matic.network/"
     },
     {
-        "name": "Matic Testnet Mumbai",
```
Comment (UncleBill) [inline packages/web3-shared/evm/assets/chains.json]: Don't remove "Matic Testnet Mumbai"

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — APPROVED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — APPROVED** [review]

**Comment by guanbinrui** [comment]
I'm closing this PR. Cause it doesn't belong to any bounty issue.

**Pull request closed without merging**